x-pack/plugins/security_solution/public/timelines/components/side_panel/event_details/flyout/index.tsx

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import type { AlertsTableFlyoutBaseProps } from '@kbn/triggers-actions-ui-plugin/public';
+import { EntityType, TimelineId } from '@kbn/timelines-plugin/common';
+import { noop, some } from 'lodash/fp';
+import React, { useCallback, useMemo, useState } from 'react';
+
+import { buildHostNamesFilter } from '../../../../../../common/search_strategy';
+import { HostRisk, useHostRiskScore } from '../../../../../risk_score/containers';
+import { useHostIsolationTools } from '../use_host_isolation_tools';
+import { FlyoutHeaderContent } from './header';
+import { FlyoutBody } from './body';
+import { FlyoutFooter } from './footer';
+import { useTimelineEventsDetails } from '../../../../containers/details';
+import { useSourcererDataView } from '../../../../../common/containers/sourcerer';
+import { SourcererScopeName } from '../../../../../common/store/sourcerer/model';
+import { getFieldValue } from '../../../../../detections/components/host_isolation/helpers';
+
+export { FlyoutBody } from './body';
+export { FlyoutHeader } from './header';
+export { FlyoutFooter } from './footer';
+
+export const useToGetInternalFlyout = () => {
+  const { browserFields, docValueFields, runtimeMappings } = useSourcererDataView(
+    SourcererScopeName.detections
+  );
+  const [alert, setAlert] = useState<{ id?: string; indexName?: string }>({
+    id: undefined,
+    indexName: undefined,
+  });
+
+  const [loading, detailsData, rawEventData, ecsData, refetchFlyoutData] = useTimelineEventsDetails(
```

Comment:
I was not able to re-use our search strategy because there is too much logic with ecs data. It might be nice to start to revert the use of ECS data  attribute and just use the fields API. I got that you need the rawEventData and that simple enough for us to get that. 

We can do some baby steps here and discuss it more to find a good compromise to make everyone happy.

Reply:
Think we need to have a shared understanding of what an acceptable end state for solutions/alerts table/timelines plugins are before that